src/js/experimental/media-rendition-listbox.js

```diff
@@ -0,0 +1,153 @@
+import MediaChromeListbox from './media-chrome-listbox.js';
+import './media-chrome-option.js';
+import { window, document } from '../utils/server-safe-globals.js';
+import { parseRenditionList } from '../utils/utils.js';
+import { MediaUIAttributes, MediaUIEvents } from '../constants.js';
+
+const slotTemplate = document.createElement('template');
+slotTemplate.innerHTML = /*html*/`
+  <style>
+    media-chrome-option {
+      white-space: var(--media-rendition-listbox-white-space, nowrap);
+    }
+  </style>
+`;
+
+/**
+ * @attr {string} mediarenditionenabled - (read-only) Set to the enabled rendition.
+ * @attr {string} mediarenditionlist - (read-only) Set to the rendition list.
+ *
+ * @cssproperty --media-rendition-listbox-white-space - `white-space` of playback rate list item.
+ */
+class MediaRenditionListbox extends MediaChromeListbox {
+  #autoOption;
+  #renditionList = [];
+
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_RENDITION_LIST,
+      MediaUIAttributes.MEDIA_RENDITION_ENABLED,
+    ];
+  }
+
+  constructor() {
+    super({ slotTemplate });
+
+    const autoOption = document.createElement('media-chrome-option');
+
+    autoOption.part.add('option');
+    autoOption.value = '';
+    autoOption.textContent = 'Auto';
+    this.#autoOption = autoOption;
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+
+    if (attrName === MediaUIAttributes.MEDIA_RENDITION_ENABLED && oldValue !== newValue) {
+      this.value = newValue;
+
+    } else if (attrName === MediaUIAttributes.MEDIA_RENDITION_LIST && oldValue !== newValue) {
+
+      this.#renditionList = parseRenditionList(newValue);
+      this.#render();
+    }
+
+    super.attributeChangedCallback(attrName, oldValue, newValue);
+  }
+
+  connectedCallback() {
+    this.addEventListener('change', this.#onChange);
+
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener('change', this.#onChange);
+
+    super.disconnectedCallback();
+  }
+
+  get mediaRenditionList() {
+    return this.#renditionList;
+  }
+
+  set mediaRenditionList(list) {
+    this.removeAttribute(MediaUIAttributes.MEDIA_RENDITION_LIST);
+
+    this.#renditionList = list;
+    this.#render();
+  }
+
+  get mediaRenditionEnabled() {
+    if (this.value) {
+      return this.mediaRenditionList.filter(({ id }) => id == this.value);
```

Comment:
This can only ever return one item right? So no need to return an array?

Reply:
this is solved with making this plural mediaRenditionsEnabled because there can be a case w/ multiple renditions enabled